beacon-chain/core/blocks/withdrawals.go

```diff
@@ -148,31 +149,60 @@ func ValidateBLSToExecutionChange(st state.ReadOnlyBeaconState, signed *ethpb.Si
 //	        next_validator_index = ValidatorIndex(next_index % len(state.validators))
 //	        state.next_withdrawal_validator_index = next_validator_index
 func ProcessWithdrawals(st state.BeaconState, executionData interfaces.ExecutionData) (state.BeaconState, error) {
+	if st.Version() >= version.EPBS {
+		IsParentBlockFull, err := st.IsParentBlockFull()
+		if err != nil {
+			return nil, errors.Wrap(err, "could not check if parent block is full")
+		}
+
+		if !IsParentBlockFull {
+			return nil, nil
+		}
+	}
+
 	expectedWithdrawals, partialWithdrawalsCount, err := st.ExpectedWithdrawals()
 	if err != nil {
 		return nil, errors.Wrap(err, "could not get expected withdrawals")
 	}
 
 	var wdRoot [32]byte
-	if executionData.IsBlinded() {
-		r, err := executionData.WithdrawalsRoot()
+	if st.Version() >= version.EPBS {
+		withdrawal_list := make([]*enginev1.Withdrawal, 0, fieldparams.MaxWithdrawalsPerPayload)
+		withdrawal_list = append(withdrawal_list, expectedWithdrawals...)
+		wdRoot, err = ssz.WithdrawalSliceRoot(withdrawal_list, fieldparams.MaxWithdrawalsPerPayload)
```

Comment:
You do not need to copy the expected withdrawals here, in the Python spec it is done because of the difference between a SSZ List object and a Python sequence.